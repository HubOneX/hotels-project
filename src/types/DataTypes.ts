export type HotelData = {
  id: string;
  hotelName: string;
  hotelAddressFirstLine: string;
  hotelAddressSecondLine: string;
  starRating: number;
  images: ImageWithAlt[];
  roomsData: RoomData[];
};

export type ImageWithAlt = {
  url: string,
  alt: string
}

export type RoomData = {
  id: string;
  name: string;
  longDescription: string;
  occupancy: Occupancy;
};

export type Occupancy = {
  maxAdults: number;
  maxChildren: number;
  maxOverall: number;
};
