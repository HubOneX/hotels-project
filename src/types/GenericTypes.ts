import { Dispatch, SetStateAction } from "react";

export type DispatchSetState<T> = Dispatch<SetStateAction<T>>;

export type Nullable<T> = T | null;
