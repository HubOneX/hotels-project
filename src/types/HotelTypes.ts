export type HotelData = {
  id: string,
  hotelName: string;
  hotelAddressFirstLine: string;
  hotelAddressSecondLine: string;
  starRating: number;
};

export type RoomData = {
  id: string,
  name: string,
  longDescription: string,
  occupancy: Occupancy,
};

export type Occupancy = {
  maxAdutls: number;
  maxChildren: number;
  maxOverall: number;
};
