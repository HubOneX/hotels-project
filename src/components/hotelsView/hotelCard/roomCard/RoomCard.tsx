import { Box, Divider, Typography } from "@mui/material";
import React, { FC } from "react";
import { RoomData } from "../../../../types/DataTypes";
import styled from "@emotion/styled";
import breakpoints from "../../../../constants/breakpoints";
import RoomInformation from "./roomInformation/RoomInformation";

type Props = { room: RoomData };

// used 'display: grid' instead of Grid MUI component for the constant 250px col for desktop view
const GridBox = styled(Box)`
  display: grid;
  grid-template-columns: 250px 1fr;
  gap: 25px;

  @media (max-width: ${breakpoints.md}px) {
    grid-template-columns: 1fr;
  }
`;

const RoomCard: FC<Props> = ({ room }) => {
  const { name, occupancy, longDescription } = room;

  return (
    <>
      <Divider sx={{ margin: "14px 0" }} />
      <GridBox data-testid="RoomCard">
        <RoomInformation name={name} occupancy={occupancy} />
        <Typography variant="body1" sx={{ width: "98%" }}>
          {longDescription}
        </Typography>
      </GridBox>
    </>
  );
};

export default RoomCard;
