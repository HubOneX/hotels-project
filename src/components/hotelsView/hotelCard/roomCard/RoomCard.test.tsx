import { render, screen } from "@testing-library/react";
import { RoomData } from "../../../../types/HotelTypes";
import RoomCard from "./RoomCard";

const defaultProps: RoomData = {
  id: "R01",
  name: "Room",
  longDescription: "description",
  occupancy: {
    maxAdutls: 2,
    maxChildren: 1,
    maxOverall: 3,
  },
};

describe("RoomCard", () => {
  it("should render room name", () => {
    render(<RoomCard {...defaultProps} />);
    const foundElement = screen.getByText("Room");
    expect(foundElement).toBeInTheDocument();
  });

  it("should render room description", () => {
    render(<RoomCard {...defaultProps} />);
    const foundElement = screen.getByText("description");
    expect(foundElement).toBeInTheDocument();
  });

  it("should render adults occupancy", () => {
    render(<RoomCard {...defaultProps} />);
    const foundElement = screen.getByText("Adults: 2");
    expect(foundElement).toBeInTheDocument();
  });

  it("should render children occupancy", () => {
    render(<RoomCard {...defaultProps} />);
    const foundElement = screen.getByText("Children: 1");
    expect(foundElement).toBeInTheDocument();
  });
});
