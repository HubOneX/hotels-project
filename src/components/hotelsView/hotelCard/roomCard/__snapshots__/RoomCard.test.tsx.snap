// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomCard should match the snapshot 1`] = `
Array [
  <hr
    className="MuiDivider-root MuiDivider-fullWidth css-1jod6p6-MuiDivider-root"
  />,
  <div
    className="MuiBox-root css-5k4vvv"
    data-testid="RoomCard"
  >
    <div
      className="MuiContainer-root MuiContainer-maxWidthLg css-pwxgjm-MuiContainer-root"
    >
      <h6
        className="MuiTypography-root MuiTypography-subtitle1 css-1f9ga5n-MuiTypography-root"
      >
        Room
      </h6>
      <h6
        className="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
      >
        Adults: 
        2
      </h6>
      <h6
        className="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
      >
        Children: 
        1
      </h6>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body1 css-18g6yeb-MuiTypography-root"
    >
      description
    </p>
  </div>,
]
`;
